delfin/drivers/hds/vsp/vspstor.py

```diff
@@ -0,0 +1,266 @@
+# Copyright 2020 The SODA Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import time
+
+import six
+from oslo_log import log
+from oslo_utils import units
+
+from delfin import exception
+from delfin.common import constants
+from delfin.drivers import driver
+from delfin.drivers.hds.vsp import consts
+from delfin.drivers.hds.vsp import rest_handler
+from delfin.drivers.utils.rest_client import RestClient
+
+LOG = log.getLogger(__name__)
+
+
+class HdsVspDriver(driver.StorageDriver):
+    POOL_STATUS_MAP = {"POLN": constants.StoragePoolStatus.NORMAL,
+                       "POLF": constants.StoragePoolStatus.ABNORMAL,
+                       "POLS": constants.StoragePoolStatus.ABNORMAL,
+                       "POLE": constants.StoragePoolStatus.OFFLINE
+                       }
+
+    TIME_PATTERN = '%Y-%m-%dT%H:%M:%S'
+
+    REFCODE_OID = '1.3.6.1.4.1.116.5.11.4.2.3'
+    DESC_OID = '1.3.6.1.4.1.116.5.11.4.2.7'
+    TRAP_TIME_OID = '1.3.6.1.4.1.116.5.11.4.2.6'
+    TRAP_DATE_OID = '1.3.6.1.4.1.116.5.11.4.2.5'
+    TRAP_NICKNAME_OID = '1.3.6.1.4.1.116.5.11.4.2.2'
+    LOCATION_OID = '1.3.6.1.4.1.116.5.11.4.2.4'
+    SECONDS_TO_MS = 1000
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.rest_client = RestClient(**kwargs)
```

Comment:
One doubt.
`HdsVspDriver` have `rest_client` and `rest_handler`, and `rest_handler` has `rest_client`, so when should `HdsVspDriver` use `rest_client` and when use `rest_handler`? What the difference between `rest_client` and `rest_handler`?

Reply:
rest_handler will use rest_client,and in HdsVspDriver, i init rest_client first and init rest_handler with it to rest_handler,it's a same rest_client

Reply:
rest_client just do rest init, http request,rest_handler do bussiness

Reply:
Yes, but for a storage driver, only one rest object is enough, two object for rest would confuse other developers.
If `rest_handler` already has `rest_client`, would it be better that not operate rest_client directly anymore?
Or we can make `rest_handler` a sub class of `rest_client`, so it can has all the method what `rest_client` has.

Reply:
changed